pyvista/core/filters/data_set.py

```diff
@@ -1100,7 +1100,10 @@ def threshold(
                 raise ValueError(
                     f'Value range must be length one for a float value or two for min/max; not ({value}).'
                 )
-            alg.ThresholdBetween(value[0], value[1])
+            # alg.ThresholdBetween(value[0], value[1])
+            alg.SetLowerThreshold(value[0])
+            alg.SetUpperThreshold(value[1])
+            alg.SetThresholdFunction(_vtk.vtkThreshold.THRESHOLD_BETWEEN)
```

Comment:
May want to put this in a separate PR to land it sooner

Reply:
Does this mean https://github.com/pyvista/pyvista/pull/2881 was an insufficient fix? Or did VTK change further? I can't figure it out, sorry.

Reply:
I think VTK changed further. This is a confusing one for sure

Reply:
This was handled in #3750